```
cd /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/tools && go build -o /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/builder -trimpath go.opentelemetry.io/collector/cmd/builder
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/builder --skip-compilation --config cmd/otelcontribcol/builder-config.yaml --output-path cmd/otelcontribcol
Flag --output-path has been deprecated, use config distribution::output_path
2024-02-27T20:54:39.623Z	INFO	internal/command.go:123	OpenTelemetry Collector Builder	{"version": "", "date": "unknown"}
2024-02-27T20:54:39.628Z	INFO	internal/command.go:159	Using config file	{"path": "cmd/otelcontribcol/builder-config.yaml"}
2024-02-27T20:54:39.628Z	INFO	builder/config.go:109	Using go	{"go-executable": "/opt/hostedtoolcache/go/1.21.7/x64/bin/go"}
2024-02-27T20:54:39.628Z	INFO	builder/main.go:67	You're building a distribution with non-aligned version of the builder. Compilation may fail due to API changes. Please upgrade your builder or API	{"builder-version": "0.94.0"}
2024-02-27T20:54:39.636Z	INFO	builder/main.go:91	Sources created	{"path": "cmd/otelcontribcol"}
2024-02-27T20:54:39.636Z	INFO	builder/main.go:25	Running go subcommand.	{"arguments": ["get", "cloud.google.com/go"]}
2024-02-27T20:54:41.684Z	INFO	builder/main.go:25	Running go subcommand.	{"arguments": ["mod", "tidy", "-compat=1.21"]}
2024-02-27T20:54:43.726Z	INFO	builder/main.go:142	Getting go modules
2024-02-27T20:54:43.726Z	INFO	builder/main.go:25	Running go subcommand.	{"arguments": ["mod", "download"]}
2024-02-27T20:54:44.331Z	INFO	builder/main.go:98	Generating source codes only, the distribution will not be compiled.
make --no-print-directory -C cmd/otelcontribcol fmt
Makefile:4: warning: overriding recipe for target 'lint'
../../Makefile.Common:199: warning: ignoring old recipe for target 'lint'
cd /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/tools && go build -o /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/goimports -trimpath golang.org/x/tools/cmd/goimports
gofmt  -w -s ./
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/goimports -w  -local github.com/open-telemetry/opentelemetry-collector-contrib ./
Generated code is out of date, please run "make genotelcontribcol" and commit the changes in this PR.
```

--- a/.chloggen/main.yaml
+++ b/.chloggen/main.yaml
@@ -1,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: awsproxyextension
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Expose service_name as configurable option. Previously, it was hardcoded as xray."
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [29550]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: [user]

--- a/extension/awsproxy/README.md
+++ b/extension/awsproxy/README.md
@@ -35,6 +35,7 @@ extensions:
     role_arn: ""
     aws_endpoint: ""
     local_mode: false
+    service_name: "xray"
 ```
 
 ### endpoint (Optional)
@@ -66,3 +67,5 @@ The IAM role used by this proxy when communicating with the AWS service. If non-
 ### aws_endpoint (Optional)
 The AWS service endpoint which this proxy forwards requests to. If not set, will default to the AWS X-Ray endpoint.
 
+### service_name (Optional)
+The AWS service name which this proxy forwards requests to. If not set, will default to "xray"

--- a/extension/awsproxy/config_test.go
+++ b/extension/awsproxy/config_test.go
@@ -44,6 +44,7 @@ func TestLoadConfig(t *testing.T) {
 					Region:      "us-west-1",
 					RoleARN:     "arn:aws:iam::123456789012:role/awesome_role",
 					AWSEndpoint: "https://another.aws.endpoint.com",
+					ServiceName: "es",
 				},
 			},
 		},

--- a/extension/awsproxy/testdata/config.yaml
+++ b/extension/awsproxy/testdata/config.yaml
@@ -8,3 +8,4 @@ awsproxy/1:
   region: "us-west-1"
   role_arn: "arn:aws:iam::123456789012:role/awesome_role"
   aws_endpoint: "https://another.aws.endpoint.com"
+  service_name: "es"

--- a/internal/aws/proxy/cfg.go
+++ b/internal/aws/proxy/cfg.go
@@ -40,6 +40,10 @@ type Config struct {
 	// will be called or not. Set to `true` to skip EC2 instance
 	// metadata check.
 	LocalMode bool `mapstructure:"local_mode"`
+
+	// ServiceName determines which service the requests are sent to.
+	// will be default to `xray`. This is mandatory for SigV4
+	ServiceName string `mapstructure:"service_name"`
 }
 
 func DefaultConfig() *Config {
@@ -55,5 +59,6 @@ func DefaultConfig() *Config {
 		Region:      "",
 		RoleARN:     "",
 		AWSEndpoint: "",
+		ServiceName: "xray",
 	}
 }

--- a/internal/aws/proxy/server.go
+++ b/internal/aws/proxy/server.go
@@ -25,7 +25,6 @@ import (
 )
 
 const (
-	service    = "xray"
 	connHeader = "Connection"
 )
 
@@ -45,13 +44,16 @@ func NewServer(cfg *Config, logger *zap.Logger) (Server, error) {
 	if cfg.ProxyAddress != "" {
 		logger.Debug("Using remote proxy", zap.String("address", cfg.ProxyAddress))
 	}
+	if cfg.ServiceName == "" {
+		cfg.ServiceName = "xray"
+	}
 
 	awsCfg, sess, err := getAWSConfigSession(cfg, logger)
 	if err != nil {
 		return nil, err
 	}
 
-	awsEndPoint, err := getServiceEndpoint(awsCfg)
+	awsEndPoint, err := getServiceEndpoint(awsCfg, cfg.ServiceName)
 	if err != nil {
 		return nil, err
 	}
@@ -101,7 +103,7 @@ func NewServer(cfg *Config, logger *zap.Logger) (Server, error) {
 			}
 
 			// Sign request. signer.Sign() also repopulates the request body.
-			_, err = signer.Sign(req, body, service, *awsCfg.Region, time.Now())
+			_, err = signer.Sign(req, body, cfg.ServiceName, *awsCfg.Region, time.Now())
 			if err != nil {
 				logger.Error("Unable to sign request", zap.Error(err))
 			}
@@ -117,13 +119,13 @@ func NewServer(cfg *Config, logger *zap.Logger) (Server, error) {
 
 // getServiceEndpoint returns X-Ray service endpoint.
 // It is guaranteed that awsCfg config instance is non-nil and the region value is non nil or non empty in awsCfg object.
-// Currently the caller takes care of it.
-func getServiceEndpoint(awsCfg *aws.Config) (string, error) {
+// Currently, the caller takes care of it.
+func getServiceEndpoint(awsCfg *aws.Config, serviceName string) (string, error) {
 	if isEmpty(awsCfg.Endpoint) {
 		if isEmpty(awsCfg.Region) {
 			return "", errors.New("unable to generate endpoint from region with nil value")
 		}
-		resolved, err := endpoints.DefaultResolver().EndpointFor(service, *awsCfg.Region, setResolverConfig())
+		resolved, err := endpoints.DefaultResolver().EndpointFor(serviceName, *awsCfg.Region, setResolverConfig())
 		return resolved.URL, err
 	}
 	return *awsCfg.Endpoint, nil

--- a/internal/aws/proxy/server_test.go
+++ b/internal/aws/proxy/server_test.go
@@ -241,7 +241,7 @@ func TestCanCreateTransport(t *testing.T) {
 }
 
 func TestGetServiceEndpointInvalidAWSConfig(t *testing.T) {
-	_, err := getServiceEndpoint(&aws.Config{})
+	_, err := getServiceEndpoint(&aws.Config{}, "")
 	assert.EqualError(t, err, "unable to generate endpoint from region with nil value")
 }
 

--- a/receiver/awsxrayreceiver/config_test.go
+++ b/receiver/awsxrayreceiver/config_test.go
@@ -51,6 +51,7 @@ func TestLoadConfig(t *testing.T) {
 					Region:      "",
 					RoleARN:     "",
 					AWSEndpoint: "",
+					ServiceName: "xray",
 				},
 			},
 		},
@@ -74,6 +75,7 @@ func TestLoadConfig(t *testing.T) {
 					RoleARN:     "arn:aws:iam::123456789012:role/awesome_role",
 					AWSEndpoint: "https://another.aws.endpoint.com",
 					LocalMode:   true,
+					ServiceName: "xray",
 				},
 			}},
 	}

